ximage/tests/labels/test_labels.py

```diff
@@ -0,0 +1,736 @@
+        )
+
+        with pytest.raises(ValueError):
+            labels.redefine_label_array(array, label_indices_with_duplicate)
+
+    apply_to_all_array_types(check, array)
+
+    # Test invalid array type
+    with pytest.raises(TypeError):
+        labels.redefine_label_array(array)
+
+
+def test_labels():
+    """Test labels. See test_get_labels for extensive tests of all arguments."""
+
+    _ = float("nan")
+    array = np.array(
+        [
+            [1, 2, _, _, 3],
+            [_, _, _, _, 4],
+            [_, _, _, _, _],
+            [5, _, _, _, _],
+            [6, 7, _, 8, _],
+        ]
+    )
+
+    # Try wrong array type
+    with pytest.raises(TypeError):
+        labels.label(array)
+
+    # Check that all arguments are passed correctly to xr_get_labels
+    data_array = xr.DataArray(array)
+    min_value = 3
+    max_value = 8
+    min_area = 2
+    max_area = 4
+    footprint = np.array(
+        [
+            [1, 1],
+            [1, 1],
+        ]
+    )
+    sort_by = "mean"
+    sort_decreasing = False
+    data_array_returned = labels.label(
+        data_array,
+        min_value_threshold=min_value,
+        max_value_threshold=max_value,
+        min_area_threshold=min_area,
+        max_area_threshold=max_area,
+        footprint=footprint,
+        sort_by=sort_by,
+        sort_decreasing=sort_decreasing,
+    )
+    labels_array_expected = np.array(
+        [
+            [_, _, _, _, 1],
+            [_, _, _, _, 1],
+            [_, _, _, _, _],
+            [2, _, _, _, _],
+            [2, 2, _, 2, _],
+        ]
+    )
+    assert np.array_equal(
+        data_array_returned.coords["label"], labels_array_expected, equal_nan=True
+    )
+
+    # Test with no label returned
+    max_value = 0
+    with pytest.raises(ValueError):
+        data_array_returned = labels.label(data_array, max_value_threshold=max_value)
+
+    # Test with Dataset
+    variable_name = "data"
+    dataset = xr.Dataset({variable_name: data_array})
+    dataset_returned = labels.label(dataset, variable_name)
+    labels_array_expected = np.array(
+        [
+            [3, 3, _, _, 2],
+            [_, _, _, _, 2],
+            [_, _, _, _, _],
+            [1, _, _, _, _],
+            [1, 1, _, 4, _],
+        ]
+    )
+    assert np.array_equal(dataset_returned.coords["label"], labels_array_expected, equal_nan=True)
+
+
+def test_highlight_label():
+    """Test highlight_label"""
+
+    _ = float("nan")
+    labels_array = xr.DataArray(
+        [
+            [_, 1, 2],
+            [3, _, 2],
+            [3, 3, _],
+        ]
+    )
+    label_name = "test_label"
+    label_id = 2
+    dataset = xr.Dataset({label_name: labels_array})
+    dataset_returned = labels.highlight_label(dataset, label_name, label_id)
+    labels_array_expected = np.array(
+        [
+            [0, 0, 2],
+            [0, 0, 2],
+            [0, 0, 0],
+        ]
+    )
+    print(dataset_returned)
```

Comment:
Could be more better placed within the assertion output on fail
```
assert np.array_equal(dataset_returned[label_name], labels_array_expected), (
    f"Returned {dataset_returned} but expected {labels_array_expected}"
)
```

Reply:
Indeed, I forgot to remove the print statement